docs/classes.rst

```diff
@@ -1012,4 +1012,87 @@ custom ``__init__``-style constructors.
           });
     }
 
+.. _custom_new:
 
+Customizing Python object creation
+----------------------------------
+
+Sometimes you might need to bind a class that can't be constructed in the
+usual way:
+
+.. code-block:: cpp
+
+   class Pet {
+     private:
+       Pet(/* ... */);
+     public:
+       static std::unique_ptr<Pet> make(std::string name);
+       void speak();
+   };
+
+You can use :cpp:func:`.def_static() <class_::def_static>` to
+produce bindings that let you write ``Pet.make("Fido")`` in Python,
+just like you would write ``Pet::make("Fido")`` in C++. But sometimes it's
+nice to provide a more Pythonic interface than that, like ``Pet("Fido")``.
+To do that, nanobind lets you override ``__new__``.
+
+Since this is a rarely-used feature in Python, let's recap. Object
+initialization in Python occurs in two phases:
+
+* the *constructor*, ``__new__``, allocates memory for the object;
+
+* the *initializer*, ``__init__``, sets up the object's initial state.
+
+So far, all the ways we've seen of binding C++ constructors
+(:cpp:struct:`nb::init\<..\>() <init>`, ``.def("__init__", ...)``)
+produce Python object *initializers*. nanobind augments these with its own
+Python object constructor, which allocates a Python object that has
+space in its memory layout for the C++ object to slot in.
+The ``__init__`` method then fills in that space by calling a C++ constructor.
+
+This split between ``__new__`` and ``__init__`` has a lot of benefits,
+including a reduction in unnecessary allocations, but it does mean that
+anything created from Python must be able to control where its
+C++ innards are stored. Sometimes, as with the example of ``Pet`` above,
+that's not feasible. In such cases, you can duck down a layer and override
```

Comment:
the expression "duck down" sounded odd to me (I was thinking that you are making some kind of reference to duck typing), perhaps "go down one level" or something of that sort?